### PR TITLE
feat: expose displayName in createInstance

### PR DESCRIPTION
**Pull request by AVaksman**
This PR is dependent on #797 
 
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -426,23 +463,23 @@ class Spanner extends GrpcService {
...
     const reqOpts = {
       parent: 'projects/' + this.projectId,
-      instanceId: shortName,
+      instanceId: displayName,
       instance: extend(
         {
           name: formattedName,
-          displayName: shortName,
+          displayName,
+          nodeCount: config.nodes || 1,
```
Comment (olavloite) [inline src/index.ts]: As far as I can tell, the `createInstance` method used to accept the node count to be specified both using the property `nodeCount` as well as `nodes`. If both were set, `nodes` would take precedence over `nodeCount`. This implementation seems to only accept `nodes`, which could break existing client code. Or am I missing something here?

File: src/index.ts [hunk 2]
```diff
@@ -65,6 +70,12 @@ export interface RequestConfig {
   reqOpts: any;
   gaxOpts?: {};
 }
+export interface CreateInstanceRequest {
```
Comment (olavloite) [inline src/index.ts]: Wouldn't it be better to let this interface extend the `IIinstance` interface, as they share almost all properties? This request only adds the `nodes` property.

Comment (AVaksman) [reply src/index.ts]: `IIinstance` interface provides misleading as well as unnecessary  properties for the purpose of `CreateInstance` function. I believe it is an error for `CreateInstanceRequest` interface to extend from `IIinstance` interface due to the following:

- `name` should not be part of the `config` param, it is passed in separately.
- `config` property is not optional.
- `nodeCount` should not be part of the `config` param. `nodeCount` is not sighted anywhere in the documentation and only erroneously sneaked in during typescript conversion when `CreateInstanceRequest` interface was set to extend `IIinstance` interface.
- `state` can not be set during instance creation, it is output only. 
- `endpointUris` can not be set during instance creation, it is output only.

File: src/index.ts (hunk 1 above)
Comment (AVaksman) [reply src/index.ts]: As per my comment above, I believe `nodeCount` became a part of `config` by an error when `CreateInstanceRequest` interface was set to extend from `IIinstance` interface. `nodeCount` is not listed in the documentation (which predated the introduction of types), so it is not legacy code. I believe it is confusing to have both `nodes` and `nodeCount` properties.

File: src/index.ts (hunk 2 above)
Comment (AVaksman) [reply src/index.ts]: Once we set default value for `nodes`, as introduced in the PR, `nodes` parameter becomes optional.

File: src/index.ts (hunk 1 above)
Comment (olavloite) [reply src/index.ts]: I agree that having both is confusing (and unnecessary), and if it was only recently introduced by the addition of types, then I agree that we should remove it. We should however flag it in the release notes to notify potential users that have used the property in the meantime.

File: src/index.ts (hunk 2 above)
Comment (olavloite) [reply src/index.ts]: Sounds reasonable, thanks for the elaborate explanation.

File: src/index.ts (hunk 1 above)
Comment (skuruppu) [reply src/index.ts:376]: Resolving this since we submitted the nodeCount change in the breaking changes PR.

**Pull request merged**